### DISCUSSION TITLE
Export Print_Summary to CSV

### DISCUSSION THREAD

**Discussion by biggielutz** [google/lightweight_mmm#205] [Q&A]
I am newer to python- having trouble exporting print_summary to a csv. I am getting a blank csv- think I need to define the dataframe but nothing is working. Thanks.

**Comment by AumGupta** [comment]
Hey @ccdccc ,

To be honest it is directly not possible to save the output of the function, as the `print_summary` function returns None. You can check the implementation [here](https://github.com/google/lightweight_mmm/blob/05defadca5587057d5c13c69aff845783f71c9ba/lightweight_mmm/lightweight_mmm.py#L398).

If you really want to do it, you can do something like below. Here, this function, `capture_print_summary_to_csv` captures the console output of print_summary, processes it into rows, and saves it to a CSV file.

```python
import io
import csv
import sys

def capture_print_summary_to_csv(lightweight_mmm_instance, csv_filename="output.csv"):
    """Captures the output of `print_summary` and saves it to a CSV file."""
    output_buffer = io.StringIO()
    original_stdout = sys.stdout  # Save the original stdout

    try:
        sys.stdout = output_buffer  # Redirect stdout to the buffer
        lightweight_mmm_instance.print_summary()  # Call the function
    finally:
        sys.stdout = original_stdout  # Restore stdout

    # Step 2: Process the captured output
    captured_output = output_buffer.getvalue()
    output_buffer.close()

    # Split the captured output into rows (you may need to adapt this for specific formats)
    rows = [line.split() for line in captured_output.strip().split("\n")]

    # Step 3: Write to a CSV file
    with open(csv_filename, "w", newline="") as csv_file:
        writer = csv.writer(csv_file)
        writer.writerows(rows)

    print(f"Output has been saved to '{csv_filename}'")
```

Example instantiation and usage (requires a properly instantiated `mcmc` object)
mcmc = ...  # Define or load your NumPyro MCMC object here
lightweight_mmm_instance = LightweightMMM(mcmc)
capture_print_summary_to_csv(lightweight_mmm_instance)

---

The above is a workaround and should be used only if you really want the output as csv.

Hope it helps.
